controllers/datadogagent/override/dependencies.go

```diff
@@ -6,33 +6,46 @@
 package override
 
 import (
+	"fmt"
+
 	"github.com/go-logr/logr"
 
+	securityv1 "github.com/openshift/api/security/v1"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/errors"
 
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	ddacomponent "github.com/DataDog/datadog-operator/controllers/datadogagent/component"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/configmap"
 	"github.com/DataDog/datadog-operator/pkg/kubernetes"
 )
 
 // Dependencies is used to override any resource/dependency settings with a v2alpha1.DatadogAgentComponentOverride.
-func Dependencies(logger logr.Logger, manager feature.ResourceManagers, overrides map[v2alpha1.ComponentName]*v2alpha1.DatadogAgentComponentOverride, ddaName, namespace string) (errs []error) {
+func Dependencies(logger logr.Logger, manager feature.ResourceManagers, dda *v2alpha1.DatadogAgent) (errs []error) {
+	overrides := dda.Spec.Override
+	namespace := dda.Namespace
+
 	for component, override := range overrides {
 		err := overrideRBAC(logger, manager, override, component, namespace)
 		if err != nil {
 			errs = append(errs, err)
 		}
 
 		// Handle custom agent configurations (datadog.yaml, cluster-agent.yaml, etc.)
-		errs = append(errs, overrideCustomConfigs(manager, override.CustomConfigurations, ddaName, namespace)...)
+		errs = append(errs, overrideCustomConfigs(manager, override.CustomConfigurations, dda.Name, namespace)...)
 
 		// Handle custom check configurations
 		errs = append(errs, overrideExtraConfigs(manager, override.ExtraConfd, namespace, v2alpha1.ExtraConfdConfigMapName, true)...)
 
 		// Handle custom check files
 		errs = append(errs, overrideExtraConfigs(manager, override.ExtraChecksd, namespace, v2alpha1.ExtraChecksdConfigMapName, false)...)
 	}
+
+	// Handle scc
+	errs = append(errs, overrideSCC(manager, dda)...)
```

Comment:
All the other overrides are inside the `for`. I think this one could follow the same pattern for consistency.